contracts/modules/royalty/policies/RoyaltyPolicyLAP.sol

```diff
@@ -93,38 +74,39 @@ contract RoyaltyPolicyLAP is
     /// @notice Constructor
     /// @param royaltyModule The RoyaltyModule address
     /// @param licensingModule The LicensingModule address
-    /// @param liquidSplitFactory The 0xSplits LiquidSplitFactory address
-    /// @param liquidSplitMain The 0xSplits LiquidSplitMain address
     /// @custom:oz-upgrades-unsafe-allow constructor
-    constructor(address royaltyModule, address licensingModule, address liquidSplitFactory, address liquidSplitMain) {
+    constructor(address royaltyModule, address licensingModule) {
         if (royaltyModule == address(0)) revert Errors.RoyaltyPolicyLAP__ZeroRoyaltyModule();
         if (licensingModule == address(0)) revert Errors.RoyaltyPolicyLAP__ZeroLicensingModule();
-        if (liquidSplitFactory == address(0)) revert Errors.RoyaltyPolicyLAP__ZeroLiquidSplitFactory();
-        if (liquidSplitMain == address(0)) revert Errors.RoyaltyPolicyLAP__ZeroLiquidSplitMain();
 
         ROYALTY_MODULE = royaltyModule;
         LICENSING_MODULE = licensingModule;
-        LIQUID_SPLIT_FACTORY = liquidSplitFactory;
-        LIQUID_SPLIT_MAIN = liquidSplitMain;
         _disableInitializers();
     }
 
-    /// @notice initializer for this implementation contract
+    /// @notice Initializer for this implementation contract
     /// @param governance The governance address
     function initialize(address governance) external initializer {
         __GovernableUpgradeable_init(governance);
         __ReentrancyGuard_init();
         __UUPSUpgradeable_init();
     }
 
-    /// @dev Set the ancestors vault implementation address
+    /// @dev Set the snapshot interval
+    /// @dev Enforced to be only callable by the protocol admin in governance
+    /// @param timestampInterval The minimum timestamp interval between snapshots
+    function setSnapshotInterval(uint256 timestampInterval) public onlyProtocolAdmin {
+        RoyaltyPolicyLAPStorage storage $ = _getRoyaltyPolicyLAPStorage();
+        $.snapshotInterval = timestampInterval;
```

Comment:
do you need to do any sanity check of timestampInterval?

Reply:
In this case, zero `TimestampInterval` is allowed in case we want to make it so further down the line. We can add a non-zero restriction if you think we will never never need it.

Reply:
what's the impact of 0 TimestampInterval?

Reply:
the more snapshots we have - the harder it is to query off-chain which ones have money to be claimed. This information is needed to claim.